### ISSUE TITLE
same release name across different namespaces doesn't work

### ISSUE THREAD

**Issue by azhang** [roboll/helmfile#1524]
In my helmfile.yaml, I have 

```yaml
releases:
- name: analytics
  namespace: staging
  chart: ./analytics

- name: analytics
  namespace: prod
  chart: ./analytics
  values:
    - values.prod.yaml
```

running `helmfile apply` returns

```
Building dependency release=analytics, chart=analytics
Building dependency release=analytics, chart=analytics

...


in ./helmfile.yaml: [command "/usr/local/bin/helm" exited with non-zero status:

PATH:
  /usr/local/bin/helm

ARGS:
  0: helm (4 bytes)
  1: dependency (10 bytes)
  2: build (5 bytes)
  3: analytics (9 bytes)

ERROR:
  exit status 1

EXIT STATUS
  1

STDERR:
  Error: unable to move current charts to tmp dir: link error: cannot rename analytics/charts to analytics/tmpcharts: rename analytics/charts analytics/tmpcharts: file exists
```

Same error occurs if I select by namespace.

    helmfile -f helmfile.yaml -l namespace=prod apply

**Comment by azhang** [comment]
Turns out it's a different issue. 

```
- name: analytics
  namespace: staging
  chart: ./analytics

- name: asdfasdfasd
  namespace: prod
  chart: ./analytics
  values:
    - values.prod.yaml
```
fails as well, so I think it's because i'm trying to deploy the same chart in multiple namespaces?

**Comment by mumoshu** [comment]
@azhang Thanks for reporting! I've seen this before - I thought this helm-dep-buidl error happens when multiple concurrent `helm dep build`s are executed on the same directory(=local chart).

I thought I had added some guard to prevent concurrent execution on the same chart, but probably there's still an edge-case? 🤔

**Comment by mumoshu** [comment]
So this may be a dupe/resurrection of #1438

**Comment by mumoshu** [comment]
@azhang Which version of Helmfile are you using?

**Comment by azhang** [comment]
I am using v0.130.1... but I don't remember if I updated it before or after submitting the issue -_- I worked around it by using environments so it only builds the local repo once per run

**Comment by mumoshu** [comment]
@azhang Thanks for confirming! I couldn't reproduce this in the most recent version of Helmfile. It is guarded by this block of code: https://github.com/roboll/helmfile/blob/ff81b2f2792ea71762b1fde7c60085f2de6cde0c/pkg/state/state.go#L1113-L1120

TL;DR; this shouldn't happen as long as you use Helmfile greater than or equal to v0.125.9, AND you have exactly the same chart path in your releases.

**Comment by mumoshu** [comment]
@azhang It would be super helpful if you could confirm that this doesn't reproduce with newer versions of Helmfile. Once confirmed, you can safely use it without the workaround 😃

**Comment by azhang** [comment]
I will try to test this out once I'm a bit further along so I don't accidentally mess up my current deploys! Thanks a bunch for the help!

**Comment by azhang** [comment]
It worked with the newer version! thanks again!
